include/sys/util.h

```diff
@@ -999,6 +1008,45 @@ u8_t u8_to_dec(char *buf, u8_t buflen, u8_t value);
 			0, Z_FOR_EACH_SWALLOW_INDEX, ;, \
 			F, fixed_arg, __VA_ARGS__)
 
+/* Set of internal macros used forGET_N_ARG of macros. */
+#define Z_GET_N_ARG(count, single_arg, ...)				\
+	UTIL_WHEN(count)						\
+	(								\
+		IF_ENABLED(count, (UTIL_OBSTRUCT(__DEBRACKET)		\
+		(							\
+			COND_CODE_1(single_arg,				\
+				(Z_GET_ARG1(__VA_ARGS__)), (__VA_ARGS__))\
+		)))							\
+		UTIL_OBSTRUCT(Z_GET_N_ARG_INDIRECT) ()			\
+		(							\
+			UTIL_DEC(count), single_arg,			\
+			Z_GET_ARGS_LESS_1(__VA_ARGS__)			\
+		)							\
+	)
+
+#define Z_GET_N_ARG_INDIRECT() Z_GET_N_ARG
+
+/**
+ * @brief Get nth argument from argument list.
+ *
+ * @param N Argument index to fetch. Counter from 1.
+ * @param ... Variable list of argments from which one argument is returned.
+ *
+ * @return Nth argument.
+ */
+#define GET_N_ARG(N, ...) UTIL_EVAL(Z_GET_N_ARG(N, 1, __VA_ARGS__))
```

Comment:
Would you consider renaming it GET_ARG_N or GET_NTH_ARG? GET_N_ARG seems like "Get n arguments" to much, GET_ARG_N sounds more like "Get argument N"

Reply:
Agree with @de-nordic, the name is confusing to me as well.

Reply:
renamed to `GET_ARG_N`. In fact it seems more natural. I actually unintentionally referred to `GET_ARG_N` in the commit message 😄